apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -0,0 +1,454 @@
+
+            const existingServices = existingReceiptByKey.services
+            // if not specified, then it is null
+            const newServices = item.services ? item.services : null
+            const servicesIsEqual = isEqual(existingServices, newServices)
+
+            const existingToPayDetails = existingReceiptByKey.toPayDetails
+            // if not specified, then it is null
+            const newToPayDetails = item.toPayDetails ? item.toPayDetails : null
+            const toPayDetailsIsEqual = isEqual(existingToPayDetails, newToPayDetails)
+
+            const shouldUpdateReceipt = !toPayIsEqual || !servicesIsEqual || !toPayDetailsIsEqual
+
+            if (shouldUpdateReceipt) {
+                item.id = existingReceiptByKey.id
+                receiptsToUpdate.push(item)
+            } else {
+                notChangedReceipts.push(item)
+            }
+        }
+    })
+
+    const newReceipts = []
+    for (const item of receiptsToAdd) {
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const newReceipt = await BillingReceipt.create(context, billingReceiptGQLInput)
+        newReceipts.push(newReceipt)
+    }
+
+    const updatedReceipts = []
+    for (const item of receiptsToUpdate) {
+        const itemId = item.id
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const updatableItem = omit(billingReceiptGQLInput, ['context', 'id'])
+        const updatedReceipt = await BillingReceipt.update(context, itemId, updatableItem)
+        updatedReceipts.push(updatedReceipt)
+    }
+
+    return { createdReceipts: newReceipts, updatedReceipts: updatedReceipts, notChangedReceipts: notChangedReceipts }
+}
+
+const RegisterBillingReceiptsService = new GQLCustomSchema('RegisterBillingReceiptsService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterBillingReceiptInput ' +
+                '{ ' +
+                    'importId: String! ' +
+
+                    'address: String! ' +
+                    'normalizedAddress: String ' +
+
+                    'accountNumber: String! ' +
+                    'unitName: String! ' + // Is going to be made optional in future
+                    'unitType: String! ' + // Is going to be made optional in future
+                    'fullName: String ' +
+
+                    'toPay: String! ' +
+                    'toPayDetails: BillingReceiptServiceToPayDetailsFieldInput ' +
+                    'services: [BillingReceiptServiceFieldInput] ' +
+
+                    'month: Int! ' +
+                    'year: Int! ' +
+
+                    'category: BillingCategoryWhereUniqueInput! ' +
+
+                    'tin: String! ' +
+                    'iec: String! ' + // Is going to be made optional in future
+                    'bic: String! ' +
+                    'bankAccount: String! ' +
```

Comment:
should it work with different languages and countries?

Reply:
Yes, it should! Gonna fix this in DOMA-4162